agent/php_error.c

```diff
@@ -645,6 +645,7 @@ void nr_php_error_cb(int type,
 nr_status_t nr_php_error_record_exception(nrtxn_t* txn,
                                           zval* exception,
                                           int priority,
+                                          bool add_to_segment,
                                           const char* prefix,
                                           zend_llist* filters TSRMLS_DC) {
```

Comment:
Is there a "most common" use case in terms of whether this value should be true or false? I did a quick scan and it looks like only one invocation uses a `false` value. You may be able to avoid a lot of the argument-specific changes by turning `nr_php_error_record_exception` into a wrapper for an internal function where all the logic lives and the value passed is always true, and a sibling wrapper that where the value is always false. For cases where we want the `true` value, nothing would have to change. For cases where we want the false value, the deviant behavior could be captured in the function name itself i.e. `nr_php_error_record_exception_no_segment`.
Both of these functions would call some kind of `nr_php_error_record_exception_ex` with the `true`/`false` logic handled in the wrapper.
I think this would have the following effects:
1. No need to change the function call where the desired value is `true`.
2. The ability to give a detailed explanation of when to call each function in the function doc description that will be the focus of the wrappers rather than buried in the argument field.
3. Ease-of-reading for anyone looking at this code later- a `true`/`false` argument is hard to understand at a glance while a function that describes what it does in its name is eas(ier) (in theory).

Reply:
It has been decided to change this. Adding additional signatures for the same function clutters the API and makes knowing what is/isn't possible when reading or writing code increasingly difficult. Further, we do not hold any other argument type to this same standard (nor should we). For example, passing `NULL` in for the `prefix` argument is just as difficult to understand as `false` to the `add_to_segment` argument, yet we are not asking for a `nr_php_error_record_exception_no_prefix `signature.